File: tools/playwright_check.mjs
import { chromium } from 'playwright-core';
import { existsSync } from 'node:fs';

const url = process.argv[2];

if (!url) {
  console.error('Usage: node tools/playwright_check.mjs <url>');
  process.exit(1);
}

const browserPaths = [
  'C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe',
  'C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe',
  'C:\\Program Files\\Microsoft\\Edge\\Application\\msedge.exe',
  'C:\\Program Files (x86)\\Microsoft\\Edge\\Application\\msedge.exe',
];

const existingPath = browserPaths.find((path) => {
  return existsSync(path);
});

if (!existingPath) {
  console.error('No local Chrome or Edge executable found.');
  process.exit(1);
}

const browser = await chromium.launch({
  executablePath: existingPath,
  headless: true,
});

const page = await browser.newPage();
await page.goto(url, { waitUntil: 'networkidle' });
console.log(await page.title());
await browser.close();

File: tools/playwright_check.py
from __future__ import annotations

import argparse

from playwright.sync_api import sync_playwright


def main() -> None:
    parser = argparse.ArgumentParser(description="Open a page with Playwright and print its title")
    parser.add_argument("url")
    args = parser.parse_args()

    with sync_playwright() as playwright:
        try:
            browser = playwright.chromium.launch(headless=True)
        except Exception:
            browser = playwright.chromium.launch(channel="chrome", headless=True)
        page = browser.new_page()
        page.goto(args.url, wait_until="networkidle")
        print(page.title())
        browser.close()


if __name__ == "__main__":
    main()
